src/app/services/wallet.service.ts

```diff
@@ -318,4 +323,22 @@ export class WalletService {
     }
     return arr1.join('');
   }
+
+  private getMinRequiredOutputs(amount: number, outputs: Output[]): Output[] {
+    outputs.sort( function(a, b) {
+      return a.hours - b.hours;
+    });
+
+    const minRequiredOutputs: Output[] = [];
+    let sumCoins = 0;
+
+    outputs.forEach(output => {
+      if (amount > sumCoins) {
```

Comment:
Any output without hours must be ignored. This, however, brings an additional problem: it is possible that the usable outputs do not have enough coins to complete the amount that the user wants to send. In this case, the entire process of sending coins must fail, in such a way that it is possible to inform the user the reason of the failure.

Reply:
Can we remove outputs without hours from the minRequiredOutputs array and calculate total outputs' amount before sending transaction? In that case we can send toaster "Not enough SKY Hours available to perform transaction" notification in case total outputs coins is less then amount that user wants to send.

Reply:
Sounds good, but I guess you have to remove them from the `outputs` array, instead of `minRequiredOutputs`.

Reply:
Do you mean that all outputs retrieved by apiService.getOutputs() except the ones with empty sky hours should be used as inputs for the current transaction (currently outputs retrieved by getMinRequiredOutputs() method are used)?

Reply:
No, It is ok as you have thought: to use only the smallest possible amount of outputs. Maybe I misunderstood your previous comment, but what I was saying is that you should probably remove the outputs without hours from the `outputs` array before entering the foreach, because if you remove the outpus without hours from the `minRequiredOutputs` array after the loop, errors are very likely to occur. However, if I misunderstood you comment, please proceed with what you have in mind, I am supposed to check it out again later anyway.

Reply:
Thank you for the prompt answers.